src/main/java/org/jitsi/jigasi/transcription/Transcriber.java

```diff
@@ -0,0 +1,550 @@
+    {
+        listeners.add(listener);
+    }
+
+    /**
+     * Remove a TranscriptionListener such that it will no longer be
+     * notified of new results
+     *
+     * @param listener the listener to remove
+     */
+    public void removeTranscriptionListener(TranscriptionListener listener)
+    {
+        listeners.remove(listener);
+    }
+
+    /**
+     * The transcriber can be used as a RawStreamListener to listen for new
+     * audio packets coming in through a MediaDevice. It will try to filter
+     * them based on the SSRC of the packet. If the SSRC does not match
+     * a participant added to the transcribed, an exception will be thrown
+     *
+     * An ExecutorService is used to offload work on the thread managing
+     * audio packets
+     *
+     * @param receiveStream the stream from which the audio was received
+     * @param buffer the containing the audio as well as meta-data
+     */
+    @Override
+    public void samplesRead(ReceiveStream receiveStream, Buffer buffer)
+    {
+        if(!isTranscribing())
+        {
+            return;
+        }
+
+        executorService.submit(() ->
+        {
+            // ReceiveStream uses unsigned longs
+            long ssrc = receiveStream.getSSRC() & 0xffffffffL;
+
+            Participant p = participants.get(ssrc);
+            if(p != null)
+            {
+                participants.get(ssrc).giveBuffer(buffer);
+            }
+            else
+            {
+                logger.warn("Reading from SSRC " + ssrc + " while it is "+
+                        "not known as a participant");
+            }
+        });
+    }
+
+    /**
+     * Get the MediaDevice this transcriber is listening to for audio
+     *
+     * @return the AudioMixerMediaDevice which should receive all audio
+     * needed to be transcribed
+     */
+    public AudioMixerMediaDevice getMediaDevice()
+    {
+        return this.mediaDevice;
+    }
+
+    /**
+     * This class describes a participant in a conference whose
+     * transcription is required. It manages the transcription if its own audio
+     * will locally buffered until enough audio is collected
+     */
+    private class Participant
+            implements TranscriptionListener
+    {
+        /**
+         * Currently assume everyone to have this locale
+         */
+        private final Locale ENGLISH_LOCALE = Locale.forLanguageTag("en-US");
+
+        /**
+         * The expected amount of bytes each buffer will have
+         */
+        private final int EXPECTED_AUDIO_LENGTH = 1920;
+
+        /**
+         * The size of the local buffer. A single packet is expected to contain
+         * 1920 bytes, so the size should be a multiple of 1920
+         */
+        private final int BUFFER_SIZE = EXPECTED_AUDIO_LENGTH * 25;
+
+        /**
+         * Whether we should buffer locally before sending
+         */
+        private final Boolean USE_LOCAL_BUFFER = true;
```

Comment:
> Perhaps make these static?
Made them static and moved the Locale up because inner classes can't have static declarations. 

> A little explanation of where the numbers come from would be useful: with webrtc we usually see 20ms opus frames which are decoded to 2 bytes per sample and 48000Hz sampling rate.
added

> Oh, and the aim is to have the buffer store 500ms worth of audio, right?
The aim is indeed to store 500ms to save bandwith. Although this might've been the reason I was seeing some errors yesterday ("The audio was not being received real-time"). Maybe it should be a bit lower